jetty-servlet/src/test/java/org/eclipse/jetty/servlet/AsyncServletIOTest.java

```diff
@@ -801,6 +804,106 @@ public void onAllDataRead() throws IOException
         }
     }
     
+    
+
+    @Test
+    public void testTimeoutWhilePendingWrite() throws Exception
+    {        
+        StringBuilder request = new StringBuilder(512);
+        request.append("POST /ctx/path5/info HTTP/1.1\r\n")
+        .append("Host: localhost\r\n")
+        .append("Content-Type: text/plain\r\n")
+        .append("Content-Length: 0\r\n")
+        .append("\r\n");
+        int port=_port;
+
+        try (Socket socket = new Socket("localhost",port))
+        {
+            socket.setSoTimeout(10000);
+            OutputStream out = socket.getOutputStream();
+            out.write(request.toString().getBytes(ISO_8859_1));
+            out.flush();
+            
+            Assert.assertTrue(_servlet5.latch.await(10,TimeUnit.SECONDS));            
+        }
+    }
+    
+    @SuppressWarnings("serial")
+    public class AsyncIOServlet5 extends HttpServlet
+    {
+        final CountDownLatch latch = new CountDownLatch(2);
```

Comment:
I would prefer the latch to be split in a `timeoutLatch` and `writeErrorLatch`, both with `count=1`.